infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -234,11 +235,19 @@ public void setupTopologyConfigId(long topologyConfigId) {
             IRetry.build(IntervalRetry.class, () -> {
                 try (TxnContext txn = corfuStore.txn(NAMESPACE)) {
                     for (LogReplicationMetadataType type : LogReplicationMetadataType.values()) {
-                        long val = Address.NON_ADDRESS;
                         if (type == LogReplicationMetadataType.TOPOLOGY_CONFIG_ID) {
-                            val = topologyConfigId;
+                            appendUpdate(txn, type, topologyConfigId);
+                        } else if (type == LogReplicationMetadataType.VERSION) {
```

Comment:
Perhaps we can skip this comment (because making this change would require re-testing) but still I would suggest putting in a TODO. 

The correct thing to do is to update the version, when the version is read. Because here it's (1) referring to a static, so it could be non-initialized (depending on when this is called) and (2) it could be deferred for as long as the topology-config-id is updated.. I think its cleaner to set it once the version is read. 

Again you might not want to do this change now, but leaving a TODO we can do it/clean it later on.